### DISCUSSION TITLE
unable to inspect the image (test/spiderplot:1.1.3): Error response from daemon: No such image: test/spiderplot:1.1.3

### DISCUSSION THREAD

**Discussion by ofschnai** [aquasecurity/trivy#5966] [Q&A]
### Question

I am getting below error while scanning the docker images stored in ACR using Trivy -

![image](https://github.com/aquasecurity/trivy/assets/121802270/2952173c-bb39-4b2d-8124-4b5635080034)

After trying many things, I am not able to figure out why this error is coming.  My YAML config looks like -

![image](https://github.com/aquasecurity/trivy/assets/121802270/9c5ddac9-09c3-448b-9fc7-74a2f0e8f928)

### Target

Container Image

### Scanner

Vulnerability

### Output Format

None

### Mode

None

**Comment by DmitriyLewen** [comment]
Hello @ofschnai 
Thanks for your report.

Trivy has 4 places to search images:
1) local docker image
2) local containerd image
3) local podman image
4) repository

If Trivy can't find images - you will see 4 errors.
`unable to inspect the image (test/spiderplot:1.1.3)` is error for local docker image. This is expected error.

If i understand correctly - you want to scan image from repository.
So You need to check 4th error - this is error when trying to scan an image from the repository.
As you can see - you need authentication.

You can try to move `docker logic` step before Trivy run. This should help.
Also you can set credentials manually using env or config file.
More info:
- https://aquasecurity.github.io/trivy/v0.48/docs/advanced/private-registries/
- https://github.com/aquasecurity/trivy-action#customizing

Regards, Dmitriy

**Comment by ofschnai** [comment]
Even after moving the docker ACR login step before trivy, it is giving same error 

![image](https://github.com/aquasecurity/trivy/assets/121802270/dcad6a77-b30a-4e2a-92ef-c1f6fa9f533b)

I am trying to login to Azure Container Registry and scan a particular image.

**Reply by DmitriyLewen** [reply]
hm... can you try to use `TRIVY_USERNAME` and `TRIVY_PASSWORD` envs for `Generate Trivy Report` step?
see - https://aquasecurity.github.io/trivy/v0.48/docs/advanced/private-registries/
